packages/core-mobile/app/hooks/networks/useTokenList.ts

```diff
@@ -0,0 +1,32 @@
+import { UseQueryResult, useQuery } from '@tanstack/react-query'
+import Config from 'react-native-config'
+import Logger from 'utils/Logger'
+
+const QUERY_KEY = ['tokenList', 'solana']
+
+const HOUR_IN_MS = 1000 * 60 * 60
+const DAY_IN_MS = 1000 * 60 * 60 * 24
+
+export const useTokenList = (
+  chainId: string
+): UseQueryResult<string[], Error> => {
+  return useQuery({
+    queryKey: [...QUERY_KEY, chainId],
+    queryFn: async () => {
+      try {
+        const response = await fetch(
+          `${Config.PROXY_URL}/tokenlist?includeSolana`
+        )
+        const data = await response.json()
+        return data[chainId]?.tokens || []
+      } catch (error) {
+        Logger.error('Failed to fetch SPL token metadata', error)
+        return []
+      }
+    },
```

Comment:
the query function here should return the raw response. you can use the select function to transform the response after (if you are using it in a react component). that way we only cache 1 response for all chain ids.